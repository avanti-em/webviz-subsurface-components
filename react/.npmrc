engine-strict=true
ignore-scripts=true
CYPRESS_INSTALL_BINARY=0