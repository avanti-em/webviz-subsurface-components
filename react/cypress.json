{
    "baseUrl": "http://localhost:6006/?path=/story/deckglmap--default",
    "video": true,
    "screenshotsFolder": "./cypress/snapshots/actual",
    "trashAssetsBeforeRuns": true,
    "env": {
        "failSilently": false,
        "ALWAYS_GENERATE_DIFF": false,
        "type": "base"
    }
}
