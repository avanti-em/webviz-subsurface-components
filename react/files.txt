cypress-visual-screenshots/comparison/defaultMap.cy-default-map-story.png
cypress-visual-screenshots/diff/defaultMap.cy-default-map-story.png
files.txt
