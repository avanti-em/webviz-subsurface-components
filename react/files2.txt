cypress-visual-screenshots/diff/defaultMap.cy-default-map-story.png
