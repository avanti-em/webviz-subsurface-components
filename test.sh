if [[ -d react/cypress/snapshots/diff ]]
then
    echo "::set-output name=docs_changed::true"
else
    echo "::set-output name=docs_changed::false"
fi