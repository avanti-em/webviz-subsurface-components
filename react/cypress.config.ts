import { defineConfig } from "cypress";
const getCompareSnapshotsPlugin = require("cypress-image-diff-js/dist/plugin");

export default defineConfig({
  component: {
    devServer: {
      framework: "create-react-app",
      bundler: "webpack",
    },
    video:false,
    setupNodeEvents(on, config) {
      getCompareSnapshotsPlugin(on, config);
    }
  },
});
