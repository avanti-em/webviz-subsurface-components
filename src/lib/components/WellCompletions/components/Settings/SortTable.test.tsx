import "@testing-library/jest-dom/extend-expect";
import { fireEvent, render, screen } from "@testing-library/react";
import "jest-styled-components";
import React from "react";
import { Wrapper } from "../../test/TestWrapper";
import SortTable from "./SortTable";

describe("test sorting the table", () => {
    it("snapshot test", () => {
        const { container } = render(Wrapper({ children: <SortTable /> }));
        expect(container.firstChild).toMatchSnapshot();
    });

    it("test to check ascending sort", async () => {
        const { container } = render(<SortTable />, {
            wrapper: Wrapper,
        });

        //fireEvent.click(screen.getByRole('menuitem', {name :'Sort/Group by Attributes'}))
        fireEvent.click(screen.getByRole("button", { name: "Ascending" }));
        await screen.findByText("Descending");
    });
});
