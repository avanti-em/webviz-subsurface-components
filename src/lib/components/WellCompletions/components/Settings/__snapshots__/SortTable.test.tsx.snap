// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test sorting the table snapshot test 1`] = `
<div
  class="MuiBox-root MuiBox-root-4 makeStyles-root-1"
>
  <table
    class="sc-eCImvq cAwzou"
  >
    <thead
      class="sc-pVTma czHsjy"
    >
      <tr
        class="sc-jrQzUz gRNlyM"
      >
        <th
          class="sc-iCfLBT FYYOR"
        >
          <div
            class="sc-furvIG haXEwx"
          >
            Sort by
          </div>
        </th>
        <th
          class="sc-iCfLBT FYYOR"
        >
          <div
            class="sc-furvIG haXEwx"
          >
            Direction
          </div>
        </th>
        <th
          class="sc-iCfLBT FYYOR"
        >
          <div
            class="sc-furvIG haXEwx"
          />
        </th>
      </tr>
    </thead>
    <tbody
      class="sc-jRQAMF"
    >
      <tr
        class="sc-jrQzUz gRNlyM"
      >
        <td
          class="sc-gKckTs cqxaQV"
        >
          <div
            class="sc-iWVLQz eCHTEl makeStyles-select-3"
          >
            <select
              class="sc-fyrnIy hsTrOV"
              id="range-display-mode-selector"
            >
              <option>
                well name
              </option>
              <option>
                stratigraphy depth
              </option>
              <option>
                earliest comp date
              </option>
              <option>
                type
              </option>
              <option>
                region
              </option>
              <option>
                user defined category
              </option>
            </select>
          </div>
        </td>
        <td
          class="sc-gKckTs cqxaQV"
        >
          <button
            aria-controls="simple-menu"
            aria-haspopup="true"
            class="sc-dkPtyc hVhMQi"
            tabindex="0"
            type="button"
          >
            <span
              class="sc-gsDJrp hVwA"
            >
              <span
                class="sc-bdvvaa cBHEoO"
              >
                Ascending
              </span>
            </span>
          </button>
        </td>
        <td
          class="sc-gKckTs cqxaQV"
        >
          <button
            aria-controls="simple-menu"
            aria-haspopup="true"
            class="sc-dkPtyc jOZxVe"
            tabindex="0"
            title="Add sorting level"
            type="button"
          >
            <span
              class="sc-gsDJrp hVwA"
            >
              <svg
                aria-hidden="true"
                class="sc-kfPtPH eNNZeU"
                fill="currentColor"
                height="24px"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="sc-fKVsgm"
                  clip-rule="evenodd"
                  d="M19 3H5a2 2 0 00-2 2v14a2 2 0 002 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V5h14v14zm-6-2h-2v-4H7v-2h4V7h2v4h4v2h-4v4z"
                  data-testid="eds-icon-path"
                  fill-rule="evenodd"
                  height="24"
                />
              </svg>
            </span>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
